cylc/uiserver/resolvers.py

```diff
@@ -379,10 +379,13 @@ async def cat_log(cls, id_: Tokens, log, info, file=None):
                     if buffer:
                         yield {'lines': list(buffer)}
                         buffer.clear()
-                    if proc.returncode not in [None, 0]:
+                    if proc.returncode is not None:
```

Comment:
The best solution I can think of to capture the "file deleted" case is stderr scraping :vomiting_face:

I.E. change the `'error'` message if the string:

```
no files remaining
```

Is in stderr. Will check with BSD tail to see what string it uses.

Reply:
So what/when should an error message be yielded? Is there anything wrong with the current approach?

Reply:
I think we were trying to make the "log file deleted" scenario more intuitive.

I think this would result in the message "no files remaining" appearing in the GUI (without the "cylc cat-log exited 0" bit) which " is reasonable, we could swap it out for "file removed" if we cared to scrape stderr which is yuck enough to be a valid reason not to.

Reply:
(latest commit changes this output to something must more end-user readable)